src/fairseq2/recipes/eval/asr.py

```diff
@@ -0,0 +1,223 @@
+    target_data = cast(SequenceData, examples["text"])
+
+    source_seqs, source_padding_mask = get_seqs_and_padding_mask(source_data)
+    target_seqs, target_padding_mask = get_seqs_and_padding_mask(target_data)
+
+    return Seq2SeqBatch(
+        source_seqs,
+        source_padding_mask,
+        target_seqs,
+        target_padding_mask,
+        examples,
+    )
+
+
+def _preprocess_example(
+    example: Example, tokenizer_name: str, device: torch.device
+) -> Example:
+    """
+    Preprocesses an individual example by converting the audio array to a PyTorch tensor
+    and encoding the text.
+
+    Args:
+        example (dict): A dictionary containing "audio" and "text" keys.
+        tokenizer_name (str): The name of the tokenizer to use.
+        device (torch.device): The device to store the tensors.
+
+    Returns:
+        dict: A dictionary with "audio" and "text" as PyTorch tensors.
+    """
+    tokenizer = load_text_tokenizer(tokenizer_name)
+    encoder = tokenizer.create_encoder(device=device)
+    audio_tensor = (
+        torch.from_numpy(example["audio"]["array"]).to(torch.float16).to(device)
+    )
+    text_tensor = encoder(example["text"].lower()).to(device)
+    return {"audio": audio_tensor, "text": text_tensor}
+
+
+def seq2seq_preprocessor(batch: Seq2SeqBatch) -> Tuple[SequenceBatch, SequenceBatch]:
+    return SequenceBatch(batch.source_seqs, batch.source_padding_mask), SequenceBatch(
+        batch.target_seqs, batch.target_padding_mask
+    )
+
+
+def postprocesser(
+    outputs: Any, targets: SequenceBatch, tokenizer: TextTokenizer
+) -> Tuple[List[str], List[str]]:
+    decoder = tokenizer.create_decoder()
+    pad_idx = tokenizer.vocab_info.pad_idx
+
+    hypotheses, _ = outputs.generate_hypotheses(pad_idx=pad_idx)
+    predictions = [decoder(item) for item in hypotheses]
+    references = [decoder(item) for item in targets.seqs.to(torch.int32)]
+
+    return predictions, references
+
+
+@hf_presets.decorator("librispeech_asr")
+def _librispeech_asr_config() -> AsrEvalConfig:
+    return AsrEvalConfig(
+        dataset_name="librispeech_asr",
+        model_name="wav2vec2_asr_base_10h",
+        split="test.other",
+        # converter=librispeech_asr_to_batch,
+    )
+
+
+def load_wav2vec2_asr_evaluator(
+    config: HFEvalConfig, output_dir: Path
+) -> HFEvaluator[Seq2SeqBatch]:
+    """
+    Load the evaluator used for downstream evaluation of the model
+    in a downstream dataset and report BLEU scores
+
+    Args:
+        config (HFEvalConfig): The configuration for the evaluation.
+        output_dir (Path): The output directory to store the evaluation results.
+
+    Returns:
+        HFEvaluator: Evaluation process results.
+    """
+    if not isinstance(config, AsrEvalConfig):
+        raise ValueError(f"Expect AsrEvalConfig, get {type(config)}")
+
+    iterable_ds = load_dataset(config.dataset_name, split=config.split, streaming=True)
+    max_samples = cast(
```

Comment:
We don't need this. If `max_samples` is None, `islice` will iterate until the end

Reply:
Done! https://github.com/facebookresearch/fairseq2/pull/708/commits/6bb31febe0b67f061bc8dca8179d0580751abc6f